src/display-x11.cc

```diff
@@ -324,39 +323,11 @@ bool display_output_x11::main_loop_wait(double t) {
       }
 
       /* update struts */
-      if ((changed != 0) && own_window_type.get(*state) == TYPE_PANEL) {
+      if ((changed != 0) && own_window_type.get(*state) == window_type::PANEL) {
         int sidenum = -1;
 
-        DBGP("%s", _(PACKAGE_NAME ": defining struts\n"));
-        fflush(stderr);
-
-        switch (text_alignment.get(*state)) {
-          case TOP_LEFT:
-          case TOP_RIGHT:
-          case TOP_MIDDLE: {
-            sidenum = 2;
-            break;
-          }
-          case BOTTOM_LEFT:
-          case BOTTOM_RIGHT:
-          case BOTTOM_MIDDLE: {
-            sidenum = 3;
-            break;
-          }
-          case MIDDLE_LEFT: {
-            sidenum = 0;
-            break;
-          }
-          case MIDDLE_RIGHT: {
-            sidenum = 1;
-            break;
-          }
-
-          case NONE:
-          case MIDDLE_MIDDLE: /* XXX What about these? */;
-        }
-
-        set_struts(sidenum);
```

Comment:
`sidenum` is an intermediate variable we do matching on later in `set_struts` as well.